lighthouse-plugin-publisher-ads/computed/ad-paint-time.js

```diff
@@ -109,18 +109,38 @@ class AdPaintTime extends ComputedMetric {
    */
   static async computeObservedMetric(data, context) {
     const iframes = getAdIframes(data);
-    if (!iframes.length) {
+    const {trace: {traceEvents}} = data;
+    const {ts: pageNavigationStart} =
+      traceEvents.find((e) => e.name == 'navigationStart') || {ts: 0};
+
+    if (!iframes.length || !pageNavigationStart) {
       return Promise.resolve({timing: -1});
     }
     const adFrameIds = new Set(iframes.map((s) => s.frame && s.frame.id));
-
-    const {trace: {traceEvents}} = data;
     const adPaintTime =
         getMinEventTime('firstContentfulPaint', traceEvents, adFrameIds) ||
         getMinEventTime('firstPaint', traceEvents, adFrameIds);
-    const {ts: pageNavigationStart} =
-      traceEvents.find((e) => e.name == 'navigationStart') || {ts: 0};
-    const timingMs = (adPaintTime - pageNavigationStart) * 1e-3;
+
+    let timingMs = 0;
+    if (adPaintTime) {
+      timingMs = (adPaintTime - pageNavigationStart) / 1000;
+    } else {
+      // If we don't find a first paint event in the trace, then fall back to
+      // the time of the first request.
```

Comment:
Thoughts on adding a constant firstRequest.endTime -> paintTime value here to estimate what the time would be if an ad actually did paint? In my opinion users shouldn't be rewarded for an ad not rendering.